eng/scripts/Language-Settings.ps1

```diff
@@ -349,8 +349,50 @@ function ValidatePackage($groupId, $artifactId, $version) {
     "${groupId}__${artifactId}__${version}"
   New-Item -ItemType Directory -Path $packageDirectory -Force | Out-Null
 
-  return (SourcePackageHasComFolder $artifactNamePrefix $packageDirectory) `
-    -and (PackageDependenciesResolve $artifactNamePrefix $packageDirectory)
+  # Add more validation by replicating as much of the docs CI process as
+  # possible
+  # https://github.com/Azure/azure-sdk-for-python/issues/20109
+  if (!$DocValidationImageId) 
+  {
+    Write-Host "Validating using mvn command directly on $artifactId."
+    FallbackValidation -artifactNamePrefix $artifactNamePrefix -workingDirectory $packageDirectory
+  } 
+  else 
+  {
+    Write-Host "Validating using $DocValidationImageId on $artifactId."
+    DockerValidation -packageName "$artifactId" -packageVersion "$version" -groupId "$groudId" `
+        -DocValidationImageId $DocValidationImageId -workingDirectory $packageDirectory
+  }
+}
+
+function FallbackValidation ($artifactNamePrefix, $workingDirectory) 
+{
+  return (SourcePackageHasComFolder $artifactNamePrefix $workingDirectory) `
+    -and (PackageDependenciesResolve $artifactNamePrefix $workingDirectory)
+}
+
+function DockerValidation($packageName, $packageVersion, $groupId, $DocValidationImageId, $workingdirectory) 
+{
+
+  docker run -v "${workingDirectory}:/workdir/out" `
+    -e TARGET_PACKAGE=$packageName -e TARGET_VERSION=$packageVersion -e TARGET_GROUP_ID=$groupId -t $DocValidationImageId 2>&1 | Out-Null
```

Comment:
Why do we not want to output the docker log for debugging in the case of validation failure?

Reply:
I am thinking to output to System.debug level as docindex pipeline scan all packages. The log will slow down the entire process.

Reply:
You could also capture the logs to a file and then output it after (use `tee log.txt` instead of `Out-Null`).

Reply:
$output will be empty here because the command is being piped to `Out-Null`.

Reply:
Made the changes. Output to log file also slow down the steps as we scan hundreds of packages every night.

We currently log out the failure one only. Let's test it for some time and see it is worth to out log to file.